schema_automator/generalizers/csv_data_generalizer.py

```diff
@@ -645,6 +645,12 @@ def infer_range(slot: dict, vals: set, types: dict, coerce=True) -> str:
         if all(isfloat(v) for v in nn_vals):
             return 'float'
         if all(is_date(v) for v in nn_vals):
+            if all(
+                not hasattr(parse(str(v)), 'hour') or
+                (parse(str(v)).hour == 0 and parse(str(v)).minute == 0 and parse(str(v)).second == 0)
```

Comment:
A datetime might validly have `0` for the hour, minute, and second. But that would be a pretty horrible example of a datetime, so this is a good heuristic.

Reply:
The `guess_is_datetime` function I suggested above could note that this is a heuristic based on that criteria.